outlook/src/taskpane/components/Project/TasksSection/TasksSection.tsx

```diff
@@ -0,0 +1,163 @@
+import * as React from "react";
+import Partner from "../../../../classes/Partner";
+import Project from "../../../../classes/Project";
+import Task from "../../../../classes/Task";
+import {ContentType, HttpVerb, sendHttpRequest} from "../../../../utils/httpRequest";
+import {_t} from "../../../../utils/Translator";
+import CollapseSection from "../../CollapseSection/CollapseSection";
+import TaskListItem from "../TaskList/TaskListItem";
+import api from "../../../api";
+import AppContext from '../../AppContext';
+import {Callout, DirectionalHint} from "office-ui-fabric-react";
+import SelectProjectDropdown from "./SelectProjectDropdown";
+
+type TasksSectionProps = {
+    partner: Partner;
+}
+
+type TasksSectionState = {
+    tasks: Task[];
+    isCollapsed: boolean;
+    isProjectCalloutOpen: boolean;
+    key: number;
+}
+
+class TasksSection extends React.Component<TasksSectionProps, TasksSectionState> {
+
+    constructor(props, context) {
+        super(props, context);
+        let isCollapsed = true;
+        if (props.partner.tasks && props.partner.tasks.length > 0) {
+            isCollapsed = false;
+        }
+        this.state = {
+            tasks: this.props.partner.tasks, isCollapsed: isCollapsed,
+            isProjectCalloutOpen: false, key: Math.random()
+        };
+    }
+
+    private toggleProjectCallout = () => {
+        this.setState({isProjectCalloutOpen: !this.state.isProjectCalloutOpen});
+    }
+
+    private closeProjectCallout = () => {
+        this.setState({isProjectCalloutOpen: false});
+    }
+
+    private onCollapseButtonClick = () => {
+        this.setState({isCollapsed: !this.state.isCollapsed});
+    };
+
+    private onProjectSelected = (project: Project): void => {
+        this.closeProjectCallout();
+        this.createTask(project);
+    }
+
+    private createTask = (project: Project) => {
+        Office.context.mailbox.item.body.getAsync(Office.CoercionType.Html, (result) => {
+            const message = result.value.split('<div id="x_appendonsend"></div>')[0]; // Remove the history and only log the most recent message.
+            const subject = Office.context.mailbox.item.subject;
+
+            const requestJson = {
```

Comment:
Strange naming, would name if taskValues or something like that (name about purpose or usage, not its technical details).

Reply:
Okay, name changed!